src/clj/rems/auth/oidc.clj

```diff
@@ -43,6 +45,33 @@
     (is (= "456" (get-userid {:sub "123" :atr "456" :fallback "78"})))
     (is (= "78" (get-userid {:fallback "78"})))))
 
+(defn get-mapped-userid [id-data]
+  (let [attrs (map keyword (:oidc-mapped-userid-attributes env))]
+    (->> (select-keys id-data attrs)
+         (keep (fn [[attr value]] (user-mappings/get-user-mapping (name attr) value)))
+         first)))
+
+(defn create-user-mapping! [id-data]
+  (let [mapped-attrs (map keyword (:oidc-mapped-userid-attributes env))
+        [attr userid] (first (select-keys id-data mapped-attrs))]
+    (user-mappings/create-user-mapping! {:from (name attr)
+                                         :from-value userid
+                                         :to-value (get-userid (apply dissoc id-data mapped-attrs))})))
+
+(defn should-map-userid?
+  "Determine if user-id should be mapped to `user_mappings` table.
+   user-id is mappable if:
+   - user-id is not already mapped,
+   - Token contains at least one attribute defined in `:oidc-mapped-userid-attributes` env variable, and
+   - Token contains at least one attribute not defined in `:oidc-mapped-userid-attributes` env variable."
```

Comment:
this is currently the only function with doc string. i'm not sure if it's useful though? the related code itself is quite short.